--- 
layout: default 
---
# PhenoCycler 

<details markdown="1"><summary> Version 2 (current) </summary>

## Version 2 (current) 

| Attribute                                           | Type      | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | Allowable Values                                                          | Required   |
|-----------------------------------------------------|-----------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------|------------|
| source_storage_duration_value                       | Numeric   | How long was the source material (parent) stored, prior to this sample being processed.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |                                                                | True       |
| time_since_acquisition_instrument_calibration_value | Numeric   | The amount of time since the acqusition instrument was last serviced by the vendor. This provides a metric for assessing drift in data capture.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |                                                                | False      |
| contributors_path                                   | Textfield | The path to the file with the ORCID IDs for all contributors of this dataset (e.g., "extras/contributors.tsv" or "./contributors.tsv"). This is an internal metadata field that is just used for ingest.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |                                                                | True       |
| data_path                                           | Textfield | The top level directory containing the raw and/or processed data. For a single dataset upload this might be "." where as for a data upload containing multiple datasets, this would be the directory name for the respective dataset. For instance, if the data is within a directory called "TEST001-RK" use syntax "/TEST001-RK/" for this field. If there are multiple directory levels, use the format "/TEST001-RK/Run1/Pass2" in which "Pass2" is the subdirectory where the single dataset's data is stored. This is an internal metadata field that is just used for ingest.                                                                                                                             |                                                                | True       |
| number_of_antibodies                                | Numeric   | Number of antibodies                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |                                                                | True       |
| number_of_channels                                  | Numeric   | Number of fluorescent channels imaged during each cycle.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |                                                                | True       |
| number_of_biomarker_imaging_rounds                  | Numeric   | Number of imaging rounds to capture the tagged biomarkers. For CODEX a biomarker imaging round consists of 1. oligo application, 2. fluor application, 3. washes. For Cell DIVE a biomarker imaging round consists of 1. staining of a biomarker via secondary detection or direct conjugate and 2. dye inactivation.                                                                                                                                                                                                                                                                                                                                                                                            |                                                                | True       |
| number_of_total_imaging_rounds                      | Numeric   | The total number of acquisitions performed on microscope to collect autofluorescence/background or stained signal (e.g., histology).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |                                                                | True       |
| slide_id                                            | Textfield | A unique ID denoting the slide used. This allows users the ability to determine which tissue sections were processed together on the same slide. It is recommended that data providers prefix the ID with the center name, to prevent values overlapping across centers.                                                                                                                                                                                                                                                                                                                                                                                                                                         |                                                                | False      |
| total_run_time_value                                | Numeric   | How long the tissue was on the acquisition instrument.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |                                                                | True       |
| dataset_type                                        | Allowable Value      | The specific type of dataset being produced.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | ```10X Multiome``` ```2D Imaging Mass Cytometry``` ```ATACseq``` ```Auto-fluorescence``` ```Cell DIVE``` ```CODEX``` ```Confocal``` ```CosMx``` ```CyCIF``` ```DBiT``` ```DESI``` ```Enhanced Stimulated Raman Spectroscopy (SRS)``` ```GeoMx (nCounter)``` ```GeoMx (NGS)``` ```HiFi-Slide``` ```Histology``` ```LC-MS``` ```Light Sheet``` ```MALDI``` ```MERFISH``` ```MIBI``` ```Molecular Cartography``` ```MUSIC``` ```nanoSPLITS``` ```PhenoCycler``` ```Resolve``` ```RNAseq``` ```RNAseq (with probes)``` ```Second Harmonic Generation (SHG)``` ```SIMS``` ```SNARE-seq2``` ```Stereo-seq``` ```Thick section Multiphoton MxIF``` ```Visium (no probes)``` ```Visium (with probes)``` ```Xenium```| True       |
| analyte_class                                       | Allowable Value      | Analytes are the target molecules being measured with the assay.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |  ```Chromatin``` ```DNA``` ```DNA + RNA``` ```Endogenous fluorophores``` ```Fluorochrome``` ```Lipid``` ```Metabolite``` ```Nucleic acid and protein``` ```Peptide``` ```Polysaccharide``` ```Protein``` ```RNA  ```| True       |
| acquisition_instrument_vendor                       | Allowable Value      | An acquisition instrument is the device that contains the signal detection hardware and signal processing software. Assays generate signals such as light of various intensities or color or signals representing the molecular mass.                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | ```Akoya Biosciences``` ```Andor``` ```BGI Genomics``` ```Bruker``` ```Cytiva``` ```Evident Scientific (Olympus)``` ```GE Healthcare``` ```Hamamatsu``` ```Huron Digital Pathology``` ```Illumina``` ```In-House``` ```Ionpath``` ```Keyence``` ```Leica Biosystems``` ```Leica Microsystems``` ```Motic``` ```NanoString``` ```Resolve Biosciences``` ```Sciex``` ```Standard BioTools (Fluidigm)``` ```Thermo Fisher Scientific``` ```Zeiss Microscopy``` | True       |
| acquisition_instrument_model                        | Allowable Value      | Manufacturers of an acquisition instrument may offer various versions (models) of that instrument with different features or sensitivities. Differences in features or sensitivities may be relevant to processing or interpretation of the data.                                                                                                                                                                                                                                                                                                                                                                                                                                                                | ```Aperio AT2``` ```Aperio CS2``` ```Axio Observer 3``` ```Axio Observer 5``` ```Axio Observer 7``` ```Axio Scan.Z1``` ```BZ-X710``` ```BZ-X800``` ```BZ-X810``` ```CosMx Spatial Molecular Imager``` ```Custom: Multiphoton``` ```Digital Spatial Profiler``` ```DM6 B``` ```DNBSEQ-T7``` ```EVOS M7000``` ```HiSeq 2500``` ```HiSeq 4000``` ```Hyperion Imaging System``` ```IN Cell Analyzer 2200``` ```Lightsheet 7``` ```MALDI timsTOF Flex Prototype``` ```MIBIscope``` ```MoticEasyScan One``` ```NanoZoomer 2.0-HT``` ```NanoZoomer S210``` ```NanoZoomer S360``` ```NanoZoomer S60``` ```NanoZoomer-SQ``` ```NextSeq 2000``` ```NextSeq 500``` ```NextSeq 550``` ```NovaSeq 6000``` ```NovaSeq X``` ```NovaSeq X Plus``` ```Orbitrap Eclipse Tribrid``` ```Orbitrap Fusion Lumos Tribrid``` ```Phenocycler-Fusion 1.0``` ```Phenocycler-Fusion 2.0``` ```PhenoImager Fusion``` ```Q Exactive``` ```Q Exactive HF``` ```Q Exactive UHMR``` ```QTRAP 5500``` ```Resolve Biosciences Molecular Cartography``` ```SCN400``` ```STELLARIS 5``` ```TissueScope LE Slide Scanner``` ```Unknown``` ```VS200 Slide Scanner``` ```Xenium Analyzer``` ```Zyla 4.2 sCMOS``` | True       |
| source_storage_duration_unit                        | Allowable Value      | The time duration unit of measurement                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | ```hour``` ```month``` ```day``` ```minute``` ```year``` | True       |
| time_since_acquisition_instrument_calibration_unit  | Allowable Value      | The time unit of measurement                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |```Column-by-column``` ```Not applicable``` ```Row-by-row``` ```Snake-by-columns``` ```Snake-by-rows``` | False      |
| total_run_time_unit                                 | Allowable Value      | The units for the total run time unit field.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | ```Hour``` ```Minute``` | True       |
| metadata_schema_id                                  | Textfield | The string that serves as the definitive identifier for the metadata schema version and is readily interpretable by computers for data validation and processing. Example: 22bc762a-5020-419d-b170-24253ed9e8d9                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |                                                                | True       |
| preparation_protocol_doi                            | Textfield      | DOI for the protocols.io page that describes the assay or sample procurment and preparation. For example for an imaging assay, the protocol might include staining of a section through the creation of an OME-TIFF file. In this case the protocol would include any image processing steps required to create the OME-TIFF file. Example: https://dx.doi.org/10.17504/protocols.io.eq2lyno9qvx9/v1                                                                                                                                                                                                                                                                                                             |                                                                | True       |
| is_targeted                                        | Allowable Value  | Specifies whether or not a specific molecule(s) is/are targeted for detection/measurement by the assay ("Yes" or "No"). The CODEX analyte is protein.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | ```Yes``` ```No```                                                     | True       |
| antibodies_path                                     | Textfield | This is the location of the antibodies.tsv file relative to the root of the top level of the upload directory structure. This path should begin with "." and would likely be something like "./extras/antibodies.tsv".                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |                                                                | True       |
| parent_sample_id                                    | Textfield | Unique SenNet or SenNet identifier of the sample (i.e., block, section or suspension) used to perform this assay. For example, for a RNAseq assay, the parent would be the suspension, whereas, for one of the imaging assays, the parent would be the tissue section. If an assay comes from multiple parent samples then this should be a comma separated list. Example: HBM386.ZGKG.235, HBM672.MKPK.442 or SNT232.UBHJ.322, SNT329.ALSK.102                                                                                                                                                                                                                                                                  |                                                                | True       |
| nuclear_marker_or_stain                             | Allowable Value  | For markers, an antibody-targetted molecule present in or around the cell nucleus, the protein or gene symbol that identifies the antibody target that is used as the nuclear marker. This symbol must match the antibody target that is either generated from the panel used or entered with custom panels. Preferably, if using a custom antibody marker, this symbol should be the HGNC symbol (https://www.genenames.org/).   For non-protein targets this is the stain name (e.g., DAPI) and, when appropriate, associated staining kit and vendor. For the PhenoCycler, this symbol must match the value found in the XPD output file.                                                                     | ```DAPI``` ```Not applicable```                                        | True       |
| cell_boundary_marker_or_stain                       | Allowable Value  | If a marker or stain was used to identify all cell boundaries in the tissue, then the name of the marker or stain should be included here. The name of the antibody-targeted molecule marker or non-antibody targeted molecule stain included here must be identical to what is found in the imaging data. For example, with the PhenoCycler, this name must match the value found in the XPD output file. If multiple marker or stains are used to identify all cell boundaries, then a comma separated list should be used here.                                                                                                                                                                               | ```NAKATPASE``` ```CD298``` ```Not applicable```                           | True       |
| non_global_files                                    | Textfield | A semicolon separated list of non-shared files to be included in the dataset. The path assumes the files are located in the "TOP/non-global/" directory. For example, for the file is TOP/non-global/lab_processed/images/1-tissue-boundary.geojson the value of this field would be "./lab_processed/images/1-tissue-boundary.geojson". After ingest, these files will be copied to the appropriate locations within the respective dataset directory tree. This field is used for internal SenNet processing. Examples for GeoMx and PhenoCycler are provided in the File Locations documentation: https://docs.google.com/document/d/1n2McSs9geA9Eli4QWQaB3c9R3wo5d5U1Xd57DWQfN5Q/edit#heading=h.1u82i4axggee |                                                                | False      |
